abci/client/local_client.go

```diff
@@ -21,7 +21,7 @@ type localClient struct {
 	Callback
 }
 
-func NewLocalClient(mtx *sync.Mutex, app types.Application) *localClient {
+func NewLocalClient(mtx *sync.Mutex, app types.Application) Client {
```

Comment:
ref relevant and related discussion here: https://github.com/tendermint/tendermint/pull/3552#discussion_r275238304